.github/workflows/publish.yml

```diff
@@ -14,25 +14,34 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
+      - uses: haya14busa/action-cond@v1
+        id: sourceeval
+        with:
+          cond: ${{ inputs.source == '' }}
+          # default nuget source. TODO change to nuget.org
+          if_true: "https://nuget.cloudsmith.io/lombiq/open-source-orchard-core-extensions/v3/index.json"
+          if_false: "${{ inputs.source == '' }}"
+      - name: Print source
+        run: echo Nuget Source:${{ steps.sourceeval.outputs.value }}
       - name: Get the version
         id: get_version
         run: |
           VERSION="${GITHUB_REF_NAME//v}"
           echo VERSION:${VERSION}
-          echo ::set-output name=VERSION::${VERSION}  
+          echo ::set-output name=VERSION::${VERSION}         
       - uses: actions/checkout@v2    
       - name: Setup .NET Core
         uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.*
       - name: Install dependencies
         run: dotnet restore -p:NuGetBuild=True
       - name: Build
-        run: dotnet build --configuration Release --no-restore -p:NuGetBuild=True
-      - name: Test
-        run: dotnet test --no-restore -p:NuGetBuild=True
+        run: dotnet build --configuration Release --no-restore -p:NuGetBuild=True -p:LangVersion=Latest
       - name: Pack
-        run: dotnet pack --output artifacts --configuration Release --no-restore --no-build  -p:NuGetBuild=True -p:Version=${{ steps.get_version.outputs.VERSION }} -p:TreatWarningsAsErrors=false 
+        run: dotnet pack --output artifacts --configuration Release --no-restore --no-build  -p:NuGetBuild=True -p:Version=${{ steps.get_version.outputs.VERSION }} -p:NoWarn=NU5104 -p:TreatWarningsAsErrors=true
       - name: Push with dotnet
-        run: dotnet nuget push artifacts/*.nupkg --api-key ${{ secrets.apikey }} --source ${{ inputs.source }} --skip-duplicate
+        run: dotnet nuget push artifacts/*.nupkg --api-key ${{ secrets.apikey }} --source ${{ steps.sourceeval.outputs.value }} --skip-duplicate
```

Comment:
Hmm, this `--skip-duplicate` really made me think about publishing these packages from the OSOCE, as described under https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/issues/23#issuecomment-989097356 (since while the build would run simultaneously for each project, they needn't have the same release cycle). Would make things easier to manage.

Reply:
I don't see how that would work, because if you publish HL from CE, then you will still have to wait for the HL version to be indexed by NuGet before you can update the HE version number and build that. So you would have to be doing conditional builds / releases.

You could pull submodules from CE, as part of the build, but the NuGet versioning / indexing thing is an issue.

It takes between 5-10 minutes (occasionally as long as half an hour) for a new version to be indexed and downloadable from NuGet, which blocks multi version CI builds.

Reply:
Do you mean that when updating HL and wanting to use it in HE, we'd need to publish HL, wait, and then publish HE? I see that but we need to do this in any case.

Reply:
Can you reply to this, please?

Reply:
Yes, when you publish HL you will need to wait before bumping the version referenced by HE before publishing HE

Reply:
OK then, thanks.